### PR TITLE
Integrating SPAdes using PyDoit

### PR DESCRIPTION
**Pull request by RVanDamme**

### CODE REVIEW

File: capture/app.py [hunk 1]
```diff
@@ -21,27 +23,44 @@ def assemble(args):
...
     mean_size = args.mean
     output = args.output
+    mem = args.memory
+    thread = args.thread
+    subsample = args.subsample
     try:
         os.makedirs(args.output)
         if args.forward and args.reverse:
-            split.split(
+            rndseed = rnd.randint(1, 1000000)
```
Comment (HadrienG) [inline capture/app.py]: I'd like to have a real fixed seed with `random.seed(1)`

File: capture/parse.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def count_record(file):
             return(tot_records)
 
 
-def parse_fq(output, file, type_f, num_sub, number_records):
+def parse_fq(output, file, type_f, num_sub, number_records, handle):
```
Comment (HadrienG) [inline capture/parse.py]: This breaks the unit testing. Please add the correct handle argument to the test suite

**Review by HadrienG — DISMISSED** [review]

**Comment by HadrienG** [comment]
I may have a faster way to select random reads. We can talk about it tomorrow afternoon

**Comment by HadrienG** [comment]
@RVanDamme Can you create unit tests for the new functions? (Granted you tested with a mock dataset and the subsampling is faster now? 😄 )

File: capture/parse.py [hunk 3]
```diff
@@ -51,58 +51,73 @@ def parse_fq(output, file, type_f, num_sub, number_records):
...
-                        sub_rec = []
+                    f"{output}/subsample_{type_f}{c_sub}.fastq",
+                    "fastq")
+                c_sub += 1
+                c += 1
+                sub_rec = []
+        else:
+            sub_rec.append(record)
+    # if there is still reads we save them in an extrafile
+    if sub_rec != []:
```
Comment (HadrienG) [inline capture/parse.py:73]: `if sub_rec` should work and is arguably better syntax

File: capture/parse.py [hunk 4]
```diff
@@ -51,58 +51,73 @@ def parse_fq(output, file, type_f, num_sub, number_records):
...
+    print(type(rnd_rec))
+    # rnd_rec = rnd_rec.sort()
+    c = 1  # saved record counter
+    rec_c = 1  # record counter
+    c_sub = 1
+    sub_rec = []
+    file_record = SeqIO.parse(handle, "fastq")
+    for record in file_record:
+        if c_sub <= num_sub:
+            if c < number_records*c_sub:
```
Comment (HadrienG) [inline capture/parse.py]: space around operators

File: capture/run_doit.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
```
Comment (HadrienG) [inline capture/run_doit.py]: calling this file `tasks` or `jobs` might be more in line with a workflow manager / the doit vocabulary

File: capture/external_run.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python
```
Comment (HadrienG) [inline capture/external_run.py]: rename `run.py`

**Review by HadrienG — CHANGES_REQUESTED** [review]

**Comment by HadrienG** [comment]
@RVanDamme ready to merge and move on to contig overlap?

**Comment by RVanDamme** [comment]
@HadrienG yes, it seems the SPAdes part is complete for now

**Review by HadrienG — APPROVED** [review]

**Pull request merged**